services/meta/client.go

```diff
@@ -60,8 +61,9 @@ type Client struct {
 }
 
 // NewClient returns a new *Client.
-func NewClient(metaServers []string, tls bool) *Client {
+func NewClient(nodeID uint64, metaServers []string, tls bool) *Client {
```

Comment:
I don't think nodeID should be in the constructor of the client. Sometimes when you create the client you won't have a node ID yet. I think it should just be an exported field that can be set.